network/src/noise/handshake.rs

```diff
@@ -248,15 +258,42 @@ impl NoiseUpgrader {
             .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
 
         // send the first handshake message
+        sl_info!(
+            network_log(network_events::NOISE_UPGRADE, &self.network_context)
+                .data("origin", "client")
+                .data("step", "writing_data")
+                .log(format!(
+                    "{} noise client writing_data: {}",
+                    self.network_context, remote_public_key
+                ))
+        );
         socket.write_all(&client_message).await?;
         socket.flush().await?;
 
         // receive the server's response (<- e, ee, se)
+        sl_info!(
+            network_log(network_events::NOISE_UPGRADE, &self.network_context)
+                .data("origin", "client")
+                .data("step", "reading_data")
+                .log(format!(
+                    "{} noise client reading_data: {}",
+                    self.network_context, remote_public_key
+                ))
+        );
         let mut server_response = [0u8; Self::SERVER_MESSAGE_SIZE];
         socket.read_exact(&mut server_response).await?;
 
         // parse the server's response
         // TODO: security logging here? (mimoo)
+        sl_info!(
+            network_log(network_events::NOISE_UPGRADE, &self.network_context)
+                .data("origin", "client")
+                .data("step", "finalizing")
+                .log(format!(
+                    "{} noise client finalize: {}",
+                    self.network_context, remote_public_key
```

Comment:
same

Reply:
this could be the response, but the response isn't in any reasonable format... I believe additional poking in this space could result in higher quality logging messages, but this should be sufficient to make sense of whether or not there are issues in establishing connections.